packages/@aws-cdk/aws-ssm/lib/parameter.ts

```diff
@@ -0,0 +1,220 @@
+   */
+  name?: string;
+}
+
+/**
+ * Properties needed to create a String SSM parameter.
+ */
+export interface StringParameterProps extends ParameterProps {
+  /**
+   * The value of the parameter. It may not reference another parameter and ``{{}}`` cannot be used in the value.
+   */
+  value: string;
+}
+
+/**
+ * Properties needed to create a StringList SSM Parameter
+ */
+export interface StringListParameterProps extends ParameterProps {
+  /**
+   * The values of the parameter. It may not reference another parameter and ``{{}}`` cannot be used in the value.
+   */
+  value: string[];
+}
+
+/**
+ * Basic features shared across all types of SSM Parameters.
+ */
+export abstract class Parameter extends cdk.Construct implements IParameter {
+  public abstract readonly parameterName: string;
+  public abstract readonly parameterType: string;
+
+  constructor(scope: cdk.Construct, id: string, _props: ParameterProps) {
+    super(scope, id);
+  }
+
+  public get parameterArn(): string {
+    return cdk.Stack.find(this).formatArn({
+      service: 'ssm',
+      resource: 'parameter',
+      resourceName: this.parameterName,
+    });
+  }
+
+  public grantRead(grantee: iam.IRole): IParameter {
+    grantee.addToPolicy(new iam.PolicyStatement()
+      .allow()
+      .addActions('ssm:DescribeParameters', 'ssm:GetParameter', 'ssm:GetParameterHistory')
+      .addResource(this.parameterArn));
+    return this;
+  }
+
+  public grantWrite(grantee: iam.IRole): IParameter {
+    grantee.addToPolicy(new iam.PolicyStatement()
+      .allow()
+      .addAction('ssm:PutParameter')
+      .addResource(this.parameterArn));
+    return this;
+  }
+}
+
+/**
+ * Creates a new String SSM Parameter.
+ */
+export class StringParameter extends Parameter implements IStringParameter {
+  public readonly parameterName: string;
+  public readonly parameterType: string;
+  public readonly parameterValue: string;
+
+  constructor(scope: cdk.Construct, id: string, props: StringParameterProps) {
+    super(scope, id, props);
+
+    if (props.allowedPattern) {
+      _assertValidValue(props.value, props.allowedPattern);
+    }
+
+    const resource = new ssm.CfnParameter(this, 'Resource', {
+      allowedPattern: props.allowedPattern,
+      description: props.description,
+      name: props.name,
+      type: 'String',
+      value: props.value,
+    });
+
+    this.parameterName = resource.parameterName;
+    this.parameterType = resource.parameterType;
+    this.parameterValue = resource.parameterValue;
+  }
+}
+
+/**
+ * Creates a new StringList SSM Parameter.
+ */
+export class StringListParameter extends Parameter implements IStringListParameter {
+  public readonly parameterName: string;
+  public readonly parameterType: string;
+  public readonly parameterValue: string[];
+
+  constructor(scope: cdk.Construct, id: string, props: StringListParameterProps) {
+    super(scope, id, props);
+
+    if (props.value.find(str => str.indexOf(',') !== -1)) {
```

Comment:
The list itself can also be a listified `Token`, in which case this check shouldn't be done either.

Reply:
It would not contain a `,`, so it'd pass, but right, skipping this in case of an unresolved token.